src/System.Runtime/tests/NSStatus/String.NSStatus.cs

```diff
@@ -0,0 +1,811 @@
+        temp = (Expected_0.Length == Local_0_1.Length); for(int i=0;i<Expected_0.Length;i++){if(Expected_0[i] != Local_0_1[i]){temp = false; break;}}
+        if(!temp) { Console.WriteLine(@"FAILED -> 0 "); errcount++;}
+        // //Summary 1-classes 1-modules 
+
+        temp = false;
+        // erid(0x8dbf69acbf253e60
+        object Param_1_0088961b = (System.String)("ToolsVersions\u005c12.0");
+        string[] Local_1_1 = ((System.String)Param_1_0088961b).Split(new char[] {'\u005c'},(System.StringSplitOptions)1);
+        string[] Expected_1 = new string[] {"ToolsVersions","12.0"};
+        temp = (Expected_1.Length == Local_1_1.Length); for(int i=0;i<Expected_1.Length;i++){if(Expected_1[i] != Local_1_1[i]){temp = false; break;}}
+        if(!temp) { Console.WriteLine(@"FAILED -> 1 "); errcount++;}
+        // //Summary 1-classes 1-modules 
+
+
+        //*** System.String.LastIndexOf-0x8188eff74bdaa0fb-101 ***
+        temp = false;
+        // erid(0x80fa43d487b85a96
+        System.String Param_2_00b650f0 = (System.String)("Microsoft.Build.Framework.ITask");
+        int Local_2_1 = Param_2_00b650f0.LastIndexOf(".",(System.StringComparison)4);
+        string Local_2_2 = Param_2_00b650f0.Substring(0,25);
+        string Local_2_3 = Param_2_00b650f0.Substring(26,5);
+        int Local_2_4 = Param_2_00b650f0.LastIndexOf(".",(System.StringComparison)4);
+        string Local_2_5 = Param_2_00b650f0.Substring(0,25);
+        string Local_2_6 = Param_2_00b650f0.Substring(26,5);
+        int Local_2_7 = Param_2_00b650f0.LastIndexOf(".",30,31,(System.StringComparison)4);
+        int Expected_2 = 25;
+        temp = (Expected_2 == Local_2_7);
+        if(!temp) { Console.WriteLine(@"FAILED -> 2 "); errcount++;}
+        // //Summary 1-classes 1-modules 
+
+
+        //*** System.String.IndexOf-0x8dc4331296d88a6e-101 ***
+        temp = false;
+        // erid(0x803b48d802354f69
+        System.String Param_3_0097a72a = (System.String)("SOFTWARE\u005cMICROSOFT\u005cMicrosoft SDKs\u005c\u005cSilverlight\u005cv5.0");
+        int Local_3_1 = Param_3_0097a72a.IndexOf("\u005c",(System.StringComparison)5);
+        int Local_3_2 = Param_3_0097a72a.IndexOf("\u005c",9,(System.StringComparison)5);
+        int Local_3_3 = Param_3_0097a72a.IndexOf("\u005c",19,(System.StringComparison)5);
+        int Local_3_4 = Param_3_0097a72a.IndexOf("\u005c",34,(System.StringComparison)5);
+        int Local_3_5 = Param_3_0097a72a.IndexOf("\u005c",35,(System.StringComparison)5);
+        int Expected_3 = 46;
+        temp = (Expected_3 == Local_3_5);
+        if(!temp) { Console.WriteLine(@"FAILED -> 3 "); errcount++;}
+        // //Summary 1-classes 1-modules 
+
+        temp = false;
+        // erid(0x804ca7b5a862a6de
+        System.String Param_4_0029dd0b = (System.String)("{A693A243-4743-4034-AED4-BEC4E79E0B3B}\u005cVisibility");
+        int Local_4_1 = Param_4_0029dd0b.IndexOf("\u005c",(System.StringComparison)5);
```

Comment:
Yes. The tooling replays the shortest sequence it observed between the creation of a string and whatever final assertion we are testing. In this case we new up the string with the GUID, call IndexOf a couple times, and then assert that we didn't find what we were looking for.